File: Assets/MercivKit/Runtime/Log.cs
using UnityEngine;

namespace MercivKit
{
    public class Log
    {
        [System.Diagnostics.Conditional("MERCIV_LOG_VERBOSE")]
        public static void Verbose(string message)
        {
            Debug.Log("C#: " + message);
        }

        public static void Info(string message)
        {
            Debug.Log("C#: " + message);
        }

        public static void Warning(string message)
        {
            Debug.LogWarning("C#: " + message);
        }

        public static void Error(string message)
        {
            Debug.LogError("C#: " + message);
        }

        public static void Exception(System.Exception exception)
        {
            Debug.LogException(exception);
        }
    }
}

File: web/src/lib/log.ts
// const VERBOSE_LOG = false
const VERBOSE_LOG = true

class Log {
    public verbose(...args: any[]) {
        if (VERBOSE_LOG) {
            console.log('SV:', ...args);
        }
    }

    public info(...args: any[]) {
        console.info('SV:', ...args);
    }

    public warn(...args: any[]) {
        console.warn('SV:', ...args);
    }

    public error(...args: any[]) {
        console.error('SV:', ...args);
    }
}

const log = new Log();
export default log;